File: app/src/main/java/pl/rpieja/flat/dto/ChargesDTO.java
package pl.rpieja.flat.dto;

import java.util.List;

/**
 * Created by radix on 2017-11-01.
 */

public class ChargesDTO {

    List<Charge> charges;
    List<Summary> summary;
    List<Expense> incomes;

    public List<Charge> getCharges() {
        return charges;
    }

    public List<Summary> getSummary() {
        return summary;
    }

    public List<Expense> getIncomes() {
        return incomes;
    }
}

File: app/src/main/java/pl/rpieja/flat/dto/ChargesDTO.kt
package pl.rpieja.flat.dto

import pl.memleak.flat.ChargesQuery

data class ChargesDTO(
        var revenues: List<Revenue>,
        var expenses: List<Expense>,
        var summary: List<Summary>
) {
    constructor(obj: ChargesQuery.Data) : this(
            revenues = obj.revenues()!!.map { Revenue(it.fragments().revenueFragment()) },
            expenses = obj.expenses()!!.map { Expense(it.fragments().expenseFragment()) },
            summary = obj.summary()!!.monthly()!!.map { Summary(it) }
    )
}
